exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/AbstractQuery.kt

```diff
@@ -31,7 +31,12 @@ abstract class AbstractQuery<T : AbstractQuery<T>>(
     /** The set of columns on which a query should be executed, contained by a [ColumnSet]. */
     abstract val set: FieldSet
 
-    protected fun copyTo(other: AbstractQuery<T>) {
+    protected abstract fun copyInstance(): T
+
+    /** Creates a new query instance using all stored properties of this query. */
+    override fun copy(): T = copyInstance().apply { copyTo(this) }
+
+    open fun copyTo(other: T) {
```

Comment:
Could you please explain the reason for introducing `copyInstance()` and overriding `SizedIterable.copy()`?
Why not just make `copyTo()` open and leave everything else the same?